### PR TITLE
feature/LT-8956_create_RecordScreenMixin

### PR DESCRIPTION
**Pull request by nguyenhoangvannha**

### CODE REVIEW

File: flutter_screen_recording/lib/mixin/record_screen_mixin.dart [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  void onDisposeScreenRecord() {
+    recordingScreen = false;
+    participantsAudioTracks.clear();
+    participantsAudioSourceNodeIds.clear();
+    recordingStatusStreamController.close();
+    recordPauseStatusStreamController.close();
+  }
+
+  Widget buildRecordingPauseStatusContainer(
+    BuildContext context,
```
Comment (DavidSonNguyen) [inline flutter_screen_recording/lib/mixin/record_screen_mixin.dart]: this does not make sense when we input context and return context in builder

Comment (nguyenhoangvannha) [reply flutter_screen_recording/lib/mixin/record_screen_mixin.dart]: -- Any parent (input context)
------StreamBuilder 
--------- builder func (stream builder context)

Input a context to remind that StreamBuilder widget will build on a context(to know where it is in the element tree).
Output context in the builder func is the context from the StreamBuilder - not from the inpur context.

File: flutter_screen_recording/lib/mixin/record_screen_mixin.dart [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import 'package:flutter_screen_recording/flutter_screen_recording.dart';
+
+mixin RecordScreenMixin {
+  //Userid, object
+  final participantsAudioTracks = <String, MediaStream>{};
+
+  //Userid, mediaStreamAudioSourceNodeId
+  final participantsAudioSourceNodeIds = <String, String>{};
+
+  final recordingStatusStreamController = StreamController<bool>()..add(false);
```
Comment (phucgaoxam) [inline flutter_screen_recording/lib/mixin/record_screen_mixin.dart]: using StreamController.broadcast() so every streams can be broadcast: https://api.dart.dev/stable/2.15.1/dart-async/StreamController/StreamController.broadcast.html

File: flutter_screen_recording/lib/mixin/record_screen_mixin.dart [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+  void storeAudioTrackReference(String uid, dynamic audioStream) {
+    participantsAudioTracks[uid] = audioStream as MediaStream;
+  }
+
+  void removeAudioTrackReference(String uid) {
+    participantsAudioTracks.remove(uid);
+  }
+
+  void addAudioTrackToScreenRecord(String uid, dynamic audioStream) {
```
Comment (phucgaoxam) [inline flutter_screen_recording/lib/mixin/record_screen_mixin.dart]: shoud avoid dynamic here

Comment (phucgaoxam) [reply flutter_screen_recording/lib/mixin/record_screen_mixin.dart]: or atleast try cach when cast type

File: flutter_screen_recording/lib/mixin/record_screen_mixin.dart (hunk 1 above)
Comment (DavidSonNguyen) [reply flutter_screen_recording/lib/mixin/record_screen_mixin.dart]: but you don't use it. And StreamBuilder is a widget, we can implement when we use, no need to have these function

Comment (DavidSonNguyen) [reply flutter_screen_recording/lib/mixin/record_screen_mixin.dart]: and don't put ui in logic class

**Review by DavidSonNguyen — APPROVED** [review]

File: flutter_screen_recording/lib/mixin/record_screen_mixin.dart [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  }
+
+  void onStopRecording(String fileName) {
+    recordingScreen = false;
+    recordingStatusStreamController.add(false);
+    participantsAudioSourceNodeIds.clear();
+  }
+
+  void storeAudioTrackReference(String uid, dynamic audioStream) {
+    participantsAudioTracks[uid] = audioStream as MediaStream;
```
Comment (phucgaoxam) [inline flutter_screen_recording/lib/mixin/record_screen_mixin.dart]: same with this one

Comment (nguyenhoangvannha) [reply flutter_screen_recording/lib/mixin/record_screen_mixin.dart]: Done

**Review by DavidSonNguyen — APPROVED** [review]

**Review by phucgaoxam — APPROVED** [review]

**Pull request merged**